src/Contracts/Connector.php

```diff
@@ -173,6 +173,14 @@ public function hasMockClient(): bool;
      */
     public function pool(iterable|callable $requests = [], int|callable $concurrency = 5, callable|null $responseHandler = null, callable|null $exceptionHandler = null): Pool;
 
+    /**
+     * @param \Saloon\Contracts\Request $request
+     * @param mixed ...$additionalArguments
+     *
+     * @return \Saloon\Contracts\RequestPaginator
+     */
+    public function paginate(Request $request, mixed ...$additionalArguments): RequestPaginator;
```

Comment:
What do you imagine we could use the $additionalArguments for?

Reply:
In case the constructor on a paginator wants more arguments.
Like.. if you want to set the limit.
However, I'm most likely removing this in favour of your other idea on extending paginators in Saloon fashion, which would defeat this purpose.
At least in general.

Reply:
Sounds good!